python/lsst/ts/MTAlignment/mock/mock_t2sa.py

```diff
@@ -0,0 +1,1044 @@
+# This file is part of ts_MTAlignment.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MockT2SA"]
+
+import asyncio
+import logging
+import re
+import typing
+
+from lsst.ts import tcpip
+from lsst.ts import utils
+from lsst.ts import salobj
+from .mock_t2sa_target import MockT2SATarget
+from ..utils import CartesianCoordinate, BodyRotation
+
+# Good reply bodies. They should come after the initial "ACK-300 " in replies,
+# and thus are intended as arguments to `write_good_reply`.
+ALREADY_MEASURING_REPLY = "ACK000"
+OK_REPLY = "ACK300"  # Command was accepted
+
+
+class MockT2SA(tcpip.OneClientServer):
+    """Emulate a New River Kinematics T2SA application.
+
+    This is a very simplistic mock with canned replys
+    for many commands.
+
+    Parameters
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCAL_HOST`
+    port : `int`
+        IP port for this server. If 0 then randomly pick an available port
+        (or ports, if listening on multiple sockets).
+        0 is strongly recommended for unit tests.
+    log : `logging.Logger`
+        Logger; if None create a new one.
+    domain : `salobj.Domain`, optional
+        Use this domain to subscribe to components telemetry and simulate
+        feedback.
+
+    Attributes
+    ----------
+    host : `str` or `None`
+        IP address for this server; typically `LOCAL_HOST`
+    port : `int`
+        IP port for this server. If this mock was constructed with port=0
+        then this will be the assigned port, once `start` is done.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+        host: str = tcpip.LOCAL_HOST,
+        port: int = 0,
+        domain: typing.Union[None, salobj.Domain] = None,
+    ) -> None:
+        self.measure_task = utils.make_done_future()
+        self.reply_loop_task = utils.make_done_future()
+
+        self._laser_warmup_task: typing.Union[None, asyncio.Task] = None
+        self.laser_warmup_status: asyncio.Future = asyncio.Future()
```

Comment:
Do you really need this? It'd be great if you just use self._laser_warmup_task instead.

For instance would something like this work:

is_warm = _laser_warmup_start_tai is not None and self._laser.warmup_task.done() and ...(unfortunately you need extra code to determine if the task succeeded, as opposed to was cancelled or failed; I really should add a function to ts_utils for that).

If you do need it:
* Please rename it to clarify the distinction and/or add a code comment
* Please end the name with "_task" (or "_future); the current name sounds to me like the value should be an enum, string, or something similar.